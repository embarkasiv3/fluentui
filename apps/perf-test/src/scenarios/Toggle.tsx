import * as React from 'react';
import { Toggle } from 'office-ui-fabric-react';

const Scenario = () => <Toggle checked />;

export default Scenario;
