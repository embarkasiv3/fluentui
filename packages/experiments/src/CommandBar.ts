// The CommandBar in @uifabric/experiments has 'graduated'.
// Consumer projects should switch to using the version from '@fluentui/react' before Fabric 7.
export * from '@fluentui/react/lib/CommandBar';
