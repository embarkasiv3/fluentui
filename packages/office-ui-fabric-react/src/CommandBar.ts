export * from './packages/react-fundamentals/CommandBar';
