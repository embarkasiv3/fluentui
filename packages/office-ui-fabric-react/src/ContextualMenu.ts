export * from './packages/react-fundamentals/ContextualMenu';
