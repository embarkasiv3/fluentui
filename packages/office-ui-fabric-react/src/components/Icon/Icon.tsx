/* tslint:disable */
import * as React from 'react';
/* tslint:enable */
import { IIconProps, IconType } from './Icon.Props';
import { Image } from '../Image/Image';
import {
  css,
  getNativeProps,
  htmlElementProperties
} from '../../Utilities';
import { getIcon, IIconRecord } from '../../Styling';
import { getClassNames } from './Icon.classNames';

export const Icon = (props: IIconProps): JSX.Element => {
  let {
    ariaLabel,
    className,
    styles,
    iconName
   } = props;
  let classNames = getClassNames(
    styles
  );

  if (props.iconType === IconType.image || props.iconType === IconType.Image) {
    let containerClassName = css(
      'ms-Icon',
      'ms-Icon-imageContainer',
      classNames.root,
      classNames.imageContainer,
      className
    );

    return (
      <div
        className={
          css(
            containerClassName,
            classNames.root
          ) }
      >
        <Image { ...props.imageProps as any } />
      </div>
    );
  } else {
    let iconDefinition = getIcon(iconName) || {
      subset: {
        className: undefined
      },
      code: undefined
    };

    return (
      <i
        aria-label={ ariaLabel }
        { ...(ariaLabel ? {} : {
          role: 'presentation',
          'aria-hidden': true,
          'data-icon-name': iconName,
        }) }
        { ...getNativeProps(props, htmlElementProperties) }
        className={
          css(
            'ms-Icon', // dangerous?
            iconDefinition.subset.className,
            classNames.root,
            props.className
          ) }
      >
        { iconDefinition.code }
      </i>
    );
  }
};
