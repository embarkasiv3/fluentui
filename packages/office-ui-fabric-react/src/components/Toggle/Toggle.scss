@import '../../common/common';

// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE in the project root for license information.

// --------------------------------------------------
// Toggle semantic slots

$toggle-background-off-color: $controlBackgroundColor;
$toggle-background-on-color: $controlBackgroundSelectedColor;
$toggle-background-on-hover-color: $controlBackgroundSelectedHoverColor;

$toggle-thumb-off-color: $controlBorderUnselectedColor;
$toggle-thumb-off-hover-color: $controlBorderUnselectedHoverColor;
$toggle-thumb-on-color: $controlForegroundSelectedColor;

$toggle-border-off-color: $toggle-thumb-off-color;
$toggle-border-off-hover-color: $toggle-thumb-off-hover-color;
$toggle-border-on-hover-color: $controlBorderSelectedHoverColor;

$toggle-thumb-off-disabled-color: $disabledTextColor;
$toggle-border-off-disabled-color: $disabledTextColor;
$toggle-background-on-disabled-color: $disabledTextColor;
$toggle-thumb-on-disabled-color: $disabledBackgroundColor;

$toggle-focusBorder-color: $focusBorderColor;

//
// Office UI Fabric
// --------------------------------------------------
// Toggle styles

@mixin highContrast-bgColor($active-color, $bw-color) {
  @media screen and (-ms-high-contrast: active) {
    background-color: $active-color;
  }

  @media screen and (-ms-high-contrast: black-on-white) {
    background-color: $bw-color;
  }
}

// Toggle
.root {
  @include ms-normalize;
  position: relative;
  display: block;
  margin-bottom: 8px;
  user-select: none;
}

// Interactive overrides
.isEnabled {

  .invisibleToggle {
    cursor: pointer;
  }

  .background {
    background: $toggle-background-off-color;
    border: 1px solid $toggle-border-off-color;
  }

  .thumb {
    background: $toggle-thumb-off-color;
    @include highContrast-bgColor($ms-color-white, $ms-color-black);
  }

  .slider:hover {
    .background {
      border: 1px solid $toggle-border-off-hover-color;
    }

    .thumb {
      background: $toggle-thumb-off-hover-color;
    }
  }

  &.isChecked {
    .background {
      background: $toggle-background-on-color;
      border: 1px solid transparent;
      @include highContrast-bgColor($ms-color-white, $ms-color-black);
    }

    .thumb {
      background: $toggle-thumb-on-color;
      @include ms-left(28px);
      @include highContrast-bgColor($ms-color-black, $ms-color-white);
    }

    .slider:hover {
      .background {
        border: 1px solid $toggle-border-on-hover-color;
        background: $toggle-background-on-hover-color;
      }

      .thumb {
        background: $toggle-thumb-on-color;
        @include left(28px);
        @include highContrast-bgColor($ms-color-black, $ms-color-white);
      }
    }
  }

}

// Disabled overrides
.isDisabled {
  .thumb {
    background: $toggle-thumb-off-disabled-color;
    @include highContrast-bgColor($ms-color-contrastBlackDisabled, $ms-color-contrastWhiteDisabled);
  }

  .background {
    background: $toggle-background-off-color;
    border: 1px solid $toggle-border-off-disabled-color;

    @media screen and (-ms-high-contrast: active) {
      border-color: $ms-color-contrastBlackDisabled;
    }

    @media screen and (-ms-high-contrast: black-on-white) {
      border-color: $ms-color-contrastWhiteDisabled;
    }
  }

  &.isChecked {
    .background {
      background: $toggle-background-on-disabled-color;
      border: 1px solid transparent;
      @include highContrast-bgColor($ms-color-contrastBlackDisabled, $ms-color-contrastWhiteDisabled);
    }

    .thumb {
      background: $toggle-thumb-on-disabled-color;
      @include ms-left(28px);
      @include highContrast-bgColor($ms-color-black, $ms-color-white);
    }
  }
}


.innerContainer {
  display: inline-block;
  min-width: 45px;
}

:global(.ms-Fabric.is-focusVisible) .root.isEnabled .invisibleToggle:focus + .background .focus {
  border: 1px solid $toggle-focusBorder-color;
}

.invisibleToggle {
  position: absolute;
  opacity: 0;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
}

.slider {
  position: relative;
  min-height: 20px;
}

.background {
  display: inline-block;
  position: absolute;
  width: 44px;
  height: 20px;
  box-sizing: border-box;
  vertical-align: middle;
  border-radius: 20px;
  cursor: pointer;
  transition: all 0.1s ease;
  pointer-events: none;
}

.thumb {
  position: absolute;
  width: 10px;
  height: 10px;
  border-radius: 10px;
  top: 4px;
  @include ms-left(4px);
  transition: all 0.1s ease;
}

.stateText {
  display: inline-block;
  vertical-align: top;
  line-height: 20px;
  @include margin-left(54px);
  padding: 0;
}

.focus {
  position: absolute;
  left: -3px;
  top: -3px;
  right: -3px;
  bottom: -3px;
  box-sizing: border-box;
  outline: transparent;
}
