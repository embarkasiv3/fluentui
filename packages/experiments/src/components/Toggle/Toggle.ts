import { ToggleView as view } from './Toggle.view';
import { ToggleStyles as styles, ToggleTokens as tokens } from './Toggle.styles';
import { ToggleState as state } from './Toggle.state';
import { IToggleProps } from './Toggle.types';
import { createComponent } from '../../Foundation';

export const Toggle: React.StatelessComponent<IToggleProps> = createComponent({
  displayName: 'Toggle',
  view,
  state,
  styles,
  tokens
});
