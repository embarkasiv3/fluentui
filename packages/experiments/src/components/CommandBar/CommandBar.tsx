import * as React from 'react';
import { styled } from '../../Utilities';
import {
  ICommandBarProps
} from './CommandBar.types';
import { CommandBarBase } from './CommandBar.base';
import { getStyles } from './CommandBar.styles';

// Create a CommandBar variant which uses these default styles and this styled subcomponent.
export const CommandBar = styled(
  CommandBarBase,
  getStyles
);