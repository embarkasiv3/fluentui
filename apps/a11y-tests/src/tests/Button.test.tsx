import * as React from 'react';
import { testComponent } from './testComponent';

import { ButtonActionExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.Action.Example';
import { ButtonAnchorExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.Anchor.Example';
import { ButtonCommandExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.Command.Example';
import { ButtonCommandBarExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.CommandBar.Example';
import { ButtonCompoundExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.Compound.Example';
import { ButtonContextualMenuExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.ContextualMenu.Example';
import { ButtonDefaultExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.Default.Example';
import { ButtonIconExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.Icon.Example';
import { ButtonPrimaryExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.Primary.Example';
import { ButtonScreenReaderExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.ScreenReader.Example';
import { ButtonSplitCustomExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.Split.Example';
import { ButtonToggleExample } from 'office-ui-fabric-react/lib/components/Button/examples/Button.Toggle.Example';

export const buttonTestComponents = [
  { name: 'Button', pageName: 'ButtonActionExample', elem: <ButtonActionExample /> },
  { name: 'Button', pageName: 'ButtonAnchorExample', elem: <ButtonAnchorExample /> },
  { name: 'Button', pageName: 'ButtonCommandExample', elem: <ButtonCommandExample /> },
  { name: 'Button', pageName: 'ButtonCommandBarExample', elem: <ButtonCommandBarExample /> },
  { name: 'Button', pageName: 'ButtonCompoundExample', elem: <ButtonCompoundExample /> },
  { name: 'Button', pageName: 'ButtonContextualMenuExample', elem: <ButtonContextualMenuExample /> },
  { name: 'Button', pageName: 'ButtonDefaultExample', elem: <ButtonDefaultExample /> },
  { name: 'Button', pageName: 'ButtonIconExample', elem: <ButtonIconExample /> },
  { name: 'Button', pageName: 'ButtonPrimaryExample', elem: <ButtonPrimaryExample /> },
  { name: 'Button', pageName: 'ButtonScreenReaderExample', elem: <ButtonScreenReaderExample /> },
  { name: 'Button', pageName: 'ButtonSplitCustomExample', elem: <ButtonSplitCustomExample /> },
  { name: 'Button', pageName: 'ButtonToggleExample', elem: <ButtonToggleExample /> }
];

buttonTestComponents.forEach(component => testComponent(component));
