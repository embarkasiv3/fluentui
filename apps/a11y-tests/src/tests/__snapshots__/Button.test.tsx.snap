// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs Axe on Button (ButtonActionExample) 1`] = `Array []`;

exports[`runs Axe on Button (ButtonAnchorExample) 1`] = `Array []`;

exports[`runs Axe on Button (ButtonCommandBarExample) 1`] = `Array []`;

exports[`runs Axe on Button (ButtonCommandExample) 1`] = `Array []`;

exports[`runs Axe on Button (ButtonCompoundExample) 1`] = `Array []`;

exports[`runs Axe on Button (ButtonContextualMenuExample) 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<button type=\\"button\\" data-automation-id=\\"test\\" class=\\"ms-Button ms-Button--default root-1\\" data-is-focusable=\\"true\\" aria-expanded=\\"false\\" aria-owns=\\"id__3-menu\\" aria-haspopup=\\"true\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "button[aria-owns=\\"id__3-menu\\"]",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- Invalid ARIA attribute value: aria-owns=\\"id__3-menu\\"",
      "text": "Fix all of the following: Invalid ARIA attribute value: aria-owns=\\"id__3-menu\\".",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "button[aria-owns=\\"id__3-menu\\"]",
      "ruleId": "aria-valid-attr-value",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-valid-attr-value",
  },
]
`;

exports[`runs Axe on Button (ButtonDefaultExample) 1`] = `Array []`;

exports[`runs Axe on Button (ButtonIconExample) 1`] = `Array []`;

exports[`runs Axe on Button (ButtonPrimaryExample) 1`] = `Array []`;

exports[`runs Axe on Button (ButtonScreenReaderExample) 1`] = `Array []`;

exports[`runs Axe on Button (ButtonSplitCustomExample) 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-10\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".css-10",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".css-10",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
]
`;

exports[`runs Axe on Button (ButtonToggleExample) 1`] = `Array []`;
